central/resourcecollection/service/service_impl.go

```diff
@@ -141,7 +155,11 @@ func collectionRequestToCollection(ctx context.Context, request collectionReques
 	if isCreate {
 		collection.CreatedBy = slimUser
 		collection.CreatedAt = timeNow
-	} else if getID != nil {
+	} else {
+		// is update
+		if getID == nil || getID() == "" {
+			return nil, errors.New("[Update collection] Valid ID getter func not found")
+		}
```

Comment:
I don't see a reason to for this. If it is an update, collection.Id must be set. Recommend focusing only on create the path and removing the else block entirely because this function does not address reconciliation with a stored collection for the update path.

Reply:
Removed this block and the getID func as well. Update workflow would need to make sure that the id is being set.